### PR TITLE
Unit test linting

### PR DESCRIPTION
**Pull request by pagarwal123**
Files updated for videojs-standard v5:

button.test.js
clickable-component.test.js
close-button.test.js
component.test.js
controls.test.js
events.test.js
extend.test.js
menu.test.js
modal-dialog.test.js
player.test.js

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: test/unit/component.test.js [hunk 1]
```diff
@@ -20,249 +22,262 @@ Component.registerComponent('TestComponent2', TestComponent2);
...
-  ok(testComp.childNameIndex_.testComponent4, 'we have a testComponent4');
+  QUnit.ok(!testComp.childNameIndex_.testComponent2, 'we do not have testComponent2');
+  QUnit.ok(testComp.childNameIndex_.testComponent4, 'we have a testComponent4');
 });
 
-test('should allows setting child options at the parent options level', function(){
-  var parent, options;
+QUnit.test('should allows setting child options at the parent options level', function() {
+  let parent;
+  let options;
```
Comment (misteroneill) [inline test/unit/component.test.js]: Can we combine this declaration and the assignment on line 189? `let options = {`

File: test/unit/clickable-component.test.js [hunk 2]
```diff
@@ -1,39 +1,44 @@
...
+  QUnit.expect(3);
 
-  let player = TestHelpers.makePlayer({});
+  const player = TestHelpers.makePlayer({});
 
-  let testClickableComponent = new ClickableComponent(player);
+  const testClickableComponent = new ClickableComponent(player);
 
-  equal(testClickableComponent.hasClass('vjs-disabled'), false, 'ClickableComponent defaults to enabled');
+  QUnit.equal(testClickableComponent.hasClass('vjs-disabled'), false,
```
Comment (misteroneill) [inline test/unit/clickable-component.test.js]: I think we ought to be consistent in the way we do function argument continuation. I believe the most common style in video.js is:

```js
QUnit.equal(testClickableComponent.hasClass('vjs-disabled'),
            false,
            'ClickableComponent defaults to enabled');
```
 I'll point out further places where this formatting should be adopted with a comment of "Function argument continuation"

File: test/unit/clickable-component.test.js [hunk 3]
```diff
@@ -1,39 +1,44 @@
...
+    'ClickableComponent defaults to enabled');
 
   testClickableComponent.disable();
 
-  equal(testClickableComponent.hasClass('vjs-disabled'), true, 'ClickableComponent is disabled');
+  QUnit.equal(testClickableComponent.hasClass('vjs-disabled'), true,
```
Comment (misteroneill) [inline test/unit/clickable-component.test.js]: Function argument continuation

File: test/unit/clickable-component.test.js [hunk 4]
```diff
@@ -1,39 +1,44 @@
...
+    'ClickableComponent is disabled');
 
   testClickableComponent.enable();
 
-  equal(testClickableComponent.hasClass('vjs-disabled'), false, 'ClickableComponent is enabled');
+  QUnit.equal(testClickableComponent.hasClass('vjs-disabled'), false,
```
Comment (misteroneill) [inline test/unit/clickable-component.test.js]: Function argument continuation

File: test/unit/close-button.test.js [hunk 5]
```diff
@@ -29,16 +31,17 @@ q.test('should create the expected element', function(assert) {
   assert.strictEqual(this.btn.el().querySelector('.vjs-control-text').innerHTML, 'Close');
 });
 
-q.test('should allow setting the controlText_ property as an option', function(assert) {
-  var text = 'OK!';
-  var btn = new CloseButton(this.player, {controlText: text});
+QUnit.test('should allow setting the controlText_ property as an option',
```
Comment (misteroneill) [inline test/unit/close-button.test.js]: Function argument continuation

File: test/unit/component.test.js [hunk 6]
```diff
@@ -20,249 +22,262 @@ Component.registerComponent('TestComponent2', TestComponent2);
...
-  ok(!Dom.hasElData(el), 'listener data nulled');
-  ok(!Object.getOwnPropertyNames(data).length, 'original listener data object was emptied');
+  QUnit.ok(hasDisposed, 'component fired dispose event');
+  QUnit.ok(bubbles === false, 'dispose event does not bubble');
+  QUnit.ok(!comp.children(), 'component children were deleted');
+  QUnit.ok(!comp.el(), 'component element was deleted');
+  QUnit.ok(!child.children(), 'child children were deleted');
+  QUnit.ok(!child.el(), 'child element was deleted');
+  QUnit.ok(!Dom.hasElData(el), 'listener data nulled');
+  QUnit.ok(!Object.getOwnPropertyNames(data).length,
```
Comment (misteroneill) [inline test/unit/component.test.js]: Function argument continuation

File: test/unit/component.test.js [hunk 7]
```diff
@@ -302,158 +318,160 @@ test('should be possible to pass data when you trigger an event', function () {
...
+  QUnit.ok(true, 'this component removed dispose listeners from other component');
 });
 
-test('should add listeners to other components and remove when them other component is disposed', function(){
-  var player = getFakePlayer(),
-      comp1 = new Component(player),
-      comp2 = new Component(player),
-      listenerFired = 0,
-      testListener;
+QUnit.test('should add listeners to other components and remove when them other component is disposed', function() {
```
Comment (misteroneill) [inline test/unit/component.test.js]: Function argument continuation

File: test/unit/component.test.js [hunk 8]
```diff
@@ -483,71 +502,71 @@ test('should not retrigger a listener when the listener calls triggerReady', fun
...
 
-  var comp = new Component(getFakePlayer(), {});
-
   comp.ready(readyListener);
   comp.triggerReady();
 
   this.clock.tick(100);
 
-  equal(timesCalled, 1, 'triggerReady from inside a ready handler does not result in an infinite loop');
+  QUnit.equal(timesCalled, 1,
```
Comment (misteroneill) [inline test/unit/component.test.js]: Function argument continuation

File: test/unit/component.test.js [hunk 9]
```diff
@@ -483,71 +502,71 @@ test('should not retrigger a listener when the listener calls triggerReady', fun
...
+  const height = 150;
+
+  const comp = new Component(getFakePlayer(), {});
   // set component dimension
 
   comp.dimensions(width, height);
 
   newWidth = comp.dimension('width', null);
 
   notEqual(newWidth, width, 'new width and old width are not the same');
```
Comment (misteroneill) [inline test/unit/component.test.js]: Looks like this should've been prefixed with `QUnit`.

Comment (misteroneill) [inline test/unit/component.test.js]: There are a couple places where this is true.

File: test/unit/component.test.js [hunk 10]
```diff
@@ -590,56 +610,61 @@ test('should get the computed dimensions', function(){
...
+  'matches computed width');
+  QUnit.equal(comp.currentDimension('height') + 'px', computedHeight,
+  'matches computed height');
 
-  equal(comp.currentDimensions()['width'] + 'px', computedWidth, 'matches computed width');
-  equal(comp.currentDimensions()['height'] + 'px', computedHeight, 'matches computed width');
+  QUnit.equal(comp.currentDimensions().width + 'px', computedWidth,
+  'matches computed width');
+  QUnit.equal(comp.currentDimensions().height + 'px', computedHeight,
+  'matches computed width');
```
Comment (misteroneill) [inline test/unit/component.test.js]: Function argument continuation - for the above 4 `equal` calls.

File: test/unit/component.test.js [hunk 11]
```diff
@@ -590,56 +610,61 @@ test('should get the computed dimensions', function(){
...
 
   comp.removeChild(child);
 
-  ok(comp.children().length === 0, 'Length should now be zero');
-  ok(comp.el().childNodes[0]['id'] === 'contentEl', 'Content El should still exist');
-  ok(comp.el().childNodes[0].childNodes[0] !== child.el(), 'Child el should be removed.');
+  QUnit.ok(comp.children().length === 0, 'Length should now be zero');
+  QUnit.ok(comp.el().childNodes[0].id === 'contentEl', 'Content El should still exist');
+  QUnit.ok(comp.el().childNodes[0].childNodes[0] !== child.el(),
+  'Child el should be removed.');
```
Comment (misteroneill) [inline test/unit/component.test.js]: Function argument continuation

File: test/unit/component.test.js [hunk 12]
```diff
@@ -720,54 +745,57 @@ test('should provide timeout methods that automatically get cleared on component
...
   ];
 
   comp.contentEl_ = contentEl;
   children.forEach(child => contentEl.appendChild(child));
 
-  strictEqual(comp.$('div'), children[0], '$ defaults to contentEl as scope');
-  strictEqual(comp.$$('div').length, children.length, '$$ defaults to contentEl as scope');
+  QUnit.strictEqual(comp.$('div'), children[0], '$ defaults to contentEl as scope');
+  QUnit.strictEqual(comp.$$('div').length, children.length,
+  '$$ defaults to contentEl as scope');
```
Comment (misteroneill) [inline test/unit/component.test.js]: Function argument continuation

File: test/unit/controls.test.js [hunk 13]
```diff
@@ -1,114 +1,122 @@
...
-  muteToggle = new MuteToggle(player);
+  const volumeControl = new VolumeControl(player);
+  const muteToggle = new MuteToggle(player);
 
-  ok(volumeControl.el().className.indexOf('vjs-hidden') >= 0, 'volumeControl is not hidden');
-  ok(muteToggle.el().className.indexOf('vjs-hidden') >= 0, 'muteToggle is not hidden');
+  QUnit.ok(volumeControl.el().className.indexOf('vjs-hidden') >= 0,
+  'volumeControl is not hidden');
+  QUnit.ok(muteToggle.el().className.indexOf('vjs-hidden') >= 0,
+  'muteToggle is not hidden');
```
Comment (misteroneill) [inline test/unit/controls.test.js]: Function argument continuation - for the above 2 `ok` calls.

File: test/unit/controls.test.js [hunk 14]
```diff
@@ -1,114 +1,122 @@
...
   for (i = 0; i < listeners.length; i++) {
     listeners[i]();
   }
 
-  equal(volumeControl.hasClass('vjs-hidden'), false, 'volumeControl does not show itself');
-  equal(muteToggle.hasClass('vjs-hidden'), false, 'muteToggle does not show itself');
+  QUnit.equal(volumeControl.hasClass('vjs-hidden'), false,
+  'volumeControl does not show itself');
+  QUnit.equal(muteToggle.hasClass('vjs-hidden'), false,
+  'muteToggle does not show itself');
```
Comment (misteroneill) [inline test/unit/controls.test.js]: Function argument continuation - in 4 places above.

File: test/unit/clickable-component.test.js (hunk 2 above)
Comment (misteroneill) [inline test/unit/clickable-component.test.js]: On second thought, all those comments seem like a bit of a waste. I'll just call out a few places.

Comment (BrandonOCasey) [inline test/unit/clickable-component.test.js]: I'm going to investigate and see if there is any way that we can get eslint to report errors/warnings on this

File: test/unit/player.test.js [hunk 15]
```diff
@@ -274,65 +284,70 @@ test('should load a media controller', function(){
...
+  // here we receive cached version instead of real
   player = videojs(videoTag.id, { techOrder: ['techFaker'] });
-  //here it triggers error, because player was destroyed already after first dispose
+  // here it triggers error, because player was destroyed already after first dispose
   player.dispose();
 });
 
 test('should set controls and trigger events', function() {
-  //expect(6);
+  // expect(6);
```
Comment (misteroneill) [inline test/unit/player.test.js]: This could just be deleted instead.

File: test/unit/player.test.js [hunk 16]
```diff
@@ -274,65 +284,70 @@ test('should load a media controller', function(){
...
-  player.on('controlsdisabled', function(){
+  player.on('controlsdisabled', function() {
     ok(true, 'disabled fired once');
   });
   player.controls(false);
-  //player.controls(true);
+  // player.controls(true);
   // Check for unnecessary events
-  //player.controls(true);
+  // player.controls(true);
```
Comment (misteroneill) [inline test/unit/player.test.js]: Let's remove the above 3 lines.

File: test/unit/clickable-component.test.js (hunk 2 above)
Comment (misteroneill) [inline test/unit/clickable-component.test.js]: 👍

File: test/unit/component.test.js (hunk 9 above)
Comment (misteroneill) [inline test/unit/component.test.js]: Actually, a bunch of places where `QUnit` functions are still being called as globals.

Comment (BrandonOCasey) [inline test/unit/component.test.js]: best way to find them is to remove the eslint-env comment at the top of the file and run `vjsstandard <file> | grep -v QUnit`

File: test/unit/component.test.js [hunk 17]
```diff
@@ -686,31 +711,31 @@ test('should emit a tap event', function(){
   browser.TOUCH_ENABLED = origTouch;
 });
 
-test('should provide timeout methods that automatically get cleared on component disposal', function() {
-  expect(4);
+QUnit.test('should provide timeout methods that automatically get cleared on component disposal',
```
Comment (BrandonOCasey) [inline test/unit/component.test.js]: you can put the function back on this line, we are going to work on ignoring the length of unit test name lines

Comment (misteroneill) [inline test/unit/component.test.js]: Yeah, good idea. We can roll back the line length changes if we want.

File: test/unit/controls.test.js [hunk 18]
```diff
@@ -1,114 +1,122 @@
...
 
-  equal(volumeControl.hasClass('vjs-hidden'), true, 'volumeControl does not hide itself');
-  equal(muteToggle.hasClass('vjs-hidden'), true, 'muteToggle does not hide itself');
+  QUnit.equal(volumeControl.hasClass('vjs-hidden'), true,
+  'volumeControl does not hide itself');
+  QUnit.equal(muteToggle.hasClass('vjs-hidden'), true, 'muteToggle does not hide itself');
 
-  player.tech_['featuresVolumeControl'] = true;
+  player.tech_.featuresVolumeControl = true;
   for (i = 0; i < listeners.length; i++) {
```
Comment (BrandonOCasey) [inline test/unit/controls.test.js]: take the `let i` from the top of this file and instead put it in this for statement `for (let i = 0;...`

File: test/unit/player.test.js [hunk 19]
```diff
@@ -43,49 +45,55 @@ test('dispose should not throw if styleEl is missing', function(){
 // Player.prototype.options_ in this file and a equivalent test using
 // videojs.options should be made in video.test.js. Keeping this here
 // until we make that move.
-test('should accept options from multiple sources and override in correct order', function(){
+test('should accept options from multiple sources and override in correct order', function() {
```
Comment (BrandonOCasey) [inline test/unit/player.test.js]: needs to be `QUnit.test`

File: test/unit/player.test.js [hunk 20]
```diff
@@ -30,8 +32,8 @@ test('should create player instance that inherits from component and dispose it'
   ok(player.el() === null, 'element disposed');
 });
 
-test('dispose should not throw if styleEl is missing', function(){
-  var player = TestHelpers.makePlayer();
+test('dispose should not throw if styleEl is missing', function() {
```
Comment (BrandonOCasey) [inline test/unit/player.test.js]: needs to be `QUnit.test`

File: test/unit/player.test.js [hunk 21]
```diff
@@ -7,21 +8,22 @@ import MediaError from '../../src/js/media-error.js';
...
   },
-  'teardown': function() {
+  teardown() {
     this.clock.restore();
   }
 });
 
-test('should create player instance that inherits from component and dispose it', function(){
-  var player = TestHelpers.makePlayer();
+test('should create player instance that inherits from component and dispose it', function() {
```
Comment (BrandonOCasey) [inline test/unit/player.test.js]: needs to be `QUnit.test`

File: test/unit/player.test.js [hunk 22]
```diff
@@ -99,16 +107,16 @@ test('should get tag, source, and track settings', function(){
...
   equal(Player.players[player.id()], player, 'player referenceable from global list');
   notEqual(tag.id, player.id, 'tag ID no longer is the same as player ID');
   notEqual(tag.className, player.el().className, 'tag classname updated');
 
   player.dispose();
 
-  notEqual(tag['player'], player, 'tag player ref killed');
-  ok(!Player.players['example_1'], 'global player ref killed');
+  notEqual(tag.player, player, 'tag player ref killed');
+  ok(!Player.players.example_1, 'global player ref killed');
```
Comment (BrandonOCasey) [inline test/unit/player.test.js]: Lots of `QUnit.` prefixes in this file

File: test/unit/player.test.js [hunk 23]
```diff
@@ -274,65 +284,65 @@ test('should load a media controller', function(){
...
+  player.on('controlsenabled', function() {
     ok(true, 'enabled fired once');
   });
-  player.on('controlsdisabled', function(){
+  player.on('controlsdisabled', function() {
     ok(true, 'disabled fired once');
   });
   player.controls(false);
-  //player.controls(true);
-  // Check for unnecessary events
```
Comment (BrandonOCasey) [inline test/unit/player.test.js]: @misternelli @gkatsev, I think we can remove all of these comments here right?

**Comment by BrandonOCasey** [comment]
LGTM other than the some minor comments and some `QUnit.` prefixes in a few files

File: test/unit/player.test.js (hunk 22 above)
Comment (pagarwal123) [inline test/unit/player.test.js]: yup...I just realized i missed the QUnit prefix for this test...i will add them

File: test/unit/component.test.js [hunk 24]
```diff
@@ -302,158 +318,160 @@ test('should be possible to pass data when you trigger an event', function () {
...
   // this component is disposed first
   listenerFired = 0;
   comp1.on(el, 'test-event', testListener);
   comp1.dispose();
   Events.trigger(el, 'test-event');
-  equal(listenerFired, 0, 'listener was removed when this component was disposed first');
-  comp1.off = function(){ throw 'Comp1 off called'; };
+  QUnit.equal(listenerFired, 0, 'listener was removed when this component was disposed first');
+  comp1.off = function() {
+    throw new Error('Comp1 off called');
```
Comment (gkatsev) [inline test/unit/component.test.js]: Thanks for making this a new Error!

File: test/unit/controls.test.js (hunk 18 above)
Comment (gkatsev) [inline test/unit/controls.test.js]: and on line 66 as well.

File: test/unit/events.test.js [hunk 25]
```diff
@@ -1,121 +1,133 @@
...
-    ok(true, 'Click Triggered');
+  const el = document.createElement('div');
+  const listener = function() {
+    QUnit.ok(true, 'Click Triggered');
   };
 
   Events.on(el, 'click', listener);
-  Events.trigger(el, 'click'); // 1 click
+  Events.trigger(el, 'click');
+  // 1 click
```
Comment (gkatsev) [inline test/unit/events.test.js]: these comments (this line and line 20) should go above the line they were associated with.

File: test/unit/player.test.js [hunk 26]
```diff
@@ -43,49 +45,55 @@ test('dispose should not throw if styleEl is missing', function(){
...
   equal(player0.options_.attr, 1, 'global option was set');
   player0.dispose();
 
   // Set a tag level option
-  let tag2 = TestHelpers.makeTag();
-  tag2.setAttribute('attr', 'asdf'); // Attributes must be set as strings
+  const tag2 = TestHelpers.makeTag();
+
+  tag2.setAttribute('attr', 'asdf');
+  // Attributes must be set as strings
```
Comment (gkatsev) [inline test/unit/player.test.js]: should go above the line

File: test/unit/player.test.js (hunk 23 above)
Comment (gkatsev) [inline test/unit/player.test.js]: it's being removed.

File: test/unit/player.test.js [hunk 27]
```diff
@@ -603,95 +621,91 @@ test('player should handle different error types', function(){
...
-    clock.restore();
+  clock.restore();
 });
 
 test('should honor configured inactivity timeout', function() {
-    var player;
-    var clock = sinon.useFakeTimers();
+  const clock = sinon.useFakeTimers();
 
     // default timeout is 2000ms, set to shorter 200ms
```
Comment (gkatsev) [inline test/unit/player.test.js]: the comments in these couple of tests are no longer aligned properly to the code.

**Comment by gkatsev** [comment]
almost there!

File: test/unit/component.test.js (hunk 24 above)
Comment (pagarwal123) [inline test/unit/component.test.js]: 👍

**Comment by gkatsev** [comment]
LGTM

**Comment by misteroneill** [comment]
LGTM

**Pull request merged**